staking/programs/integrity-pool/tests/staking_slash.rs

```diff
@@ -0,0 +1,174 @@
+    let (stake_account_metadata, _) = get_stake_account_metadata_address(stake_account_positions);
+    let (stake_account_custody, _) = get_stake_account_custody_address(stake_account_positions);
+    let (stake_account_authority, _) =
+        get_stake_account_custory_authority_address(stake_account_positions);
+    let pool_authority = Keypair::new();
+
+    let slash_token_account = create_token_account(&mut svm, &payer, &pyth_token_mint.pubkey());
+
+    update_pool_authority(&mut svm, &payer, pool_authority.pubkey());
+
+    create_position(
+        &mut svm,
+        &payer,
+        stake_account_positions,
+        staking::state::positions::TargetWithParameters::IntegrityPool {
+            publisher: publisher_keypair.pubkey(),
+        },
+        Some(&pool_authority),
+        50 * FRAC_64_MULTIPLIER,
+    );
+    create_position(
+        &mut svm,
+        &payer,
+        stake_account_positions,
+        staking::state::positions::TargetWithParameters::Voting,
+        None,
+        40 * FRAC_64_MULTIPLIER,
+    );
+    svm.expire_blockhash();
+
+    create_position(
+        &mut svm,
+        &payer,
+        stake_account_positions,
+        staking::state::positions::TargetWithParameters::Voting,
+        None,
+        40 * FRAC_64_MULTIPLIER,
+    );
+
+
+    // initiate delegate at epoch N
+    // position will become LOCKED at epoch N+1
+    // at epoch N+2, we can slash epoch N+1
+    advance_n_epochs(&mut svm, &payer, 2);
+
+    let slash_account_data = staking::instruction::SlashAccount {
+        slash_ratio:            FRAC_64_MULTIPLIER / 2,
+        target_with_parameters: staking::state::positions::TargetWithParameters::IntegrityPool {
+            publisher: publisher_keypair.pubkey(),
+        },
+    };
+
+    let (target_account, _) = get_target_address(Target::Voting);
+
+    let slash_account_accs = staking::accounts::SlashAccount {
+        config: config_pubkey,
+        stake_account_positions,
+        stake_account_metadata,
+        stake_account_custody,
+        pool_authority: pool_authority.pubkey(),
+        governance_target_account: target_account,
+        custody_authority: stake_account_authority,
+        token_program: spl_token::ID,
+        destination: slash_token_account.pubkey(),
+    };
+
+    let slash_account_ix = Instruction::new_with_bytes(
+        staking::ID,
+        &slash_account_data.data(),
+        slash_account_accs.to_account_metas(None),
+    );
+
+    let slash_account_tx = Transaction::new_signed_with_payer(
+        &[
+            slash_account_ix,
+            ComputeBudgetInstruction::set_compute_unit_limit(1_400_000),
+        ],
+        Some(&payer.pubkey()),
+        &[&payer, &pool_authority],
+        svm.latest_blockhash(),
+    );
+
+    svm.send_transaction(slash_account_tx).unwrap();
+
+    let positions: staking::state::positions::PositionData =
+        fetch_account_data_bytemuck(&mut svm, &stake_account_positions);
```

Comment:
check that there's only 2 positions left, using `next_index_`

Reply:
done.